hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/security/x509/certificate/client/DefaultCertificateClient.java

```diff
@@ -774,6 +797,19 @@ protected InitResponse handleCase(InitCase init)
     }
   }
 
+  protected void removeMaterial() throws CertificateException {
+    try {
+      FileUtils.deleteDirectory(
+          securityConfig.getKeyLocation(component).toFile());
```

Comment:
Are these directories only containing only exactly the expired public/private key and the certificate itself?

Reply:
For keys yes, for certs the dir contains also the root and intermediate CA certificates and also in case of DNs the token issuer certificates, but all of this is available via the SCM, and will be re-downloaded with normal operations like when the first initialization of  these are happening, so we don't need to care.